### ISSUE TITLE
There is a problem that needs to be solved to use the flat buffer in vcode extension

### ISSUE THREAD

**Issue by CODe1995** [Samsung/ONE-vscode#136]
## What
![image](https://user-images.githubusercontent.com/47562093/134654914-4c4c2f08-b776-4f53-8ce0-8b0f75018926.png)
We will use "require" in the process of converting typescript to javascript. However, "require" is a feature of nodejs, so it cannot be used on the client side.

## How
I found a way after searching and found that there is a [solution](https://newbedev.com/client-on-node-js-uncaught-referenceerror-require-is-not-defined) using module bundler.

There is a way to combine them into one js using browserify or [webpack](https://webpack.js.org/).
However, in this case, it will be combined into a file of 8800 lines. In this case, it is difficult to explain the code as all Flatbuffers related codes are included in the Pull Request.

For convenience, I think the most efficient method is to combine them into one file using browserify or webpack.

What would be a good way?

**Comment by seanshpark** [comment]
I understand you have a problem using `require` which is for nodejs but your execution envionment is with HTML so there is no `require` available.

But why do you need to use this feature?

**Comment by CODe1995** [comment]
Flatbuffers are used in the process of processing and visualizing circle files (#66).

![image](https://user-images.githubusercontent.com/47562093/134657427-0727aa17-88ba-45f8-8609-1e398f2c20cf.png)
The problem arises because all files associated with circle flatbuffers are automatically using "require".

**Comment by seanshpark** [comment]
> Flatbuffers are used in the process of processing and visualizing circle files 

Why do you need this? Are you making something like netron?

**Comment by CODe1995** [comment]
exactly. I am trying to show the difference in duration by comparing the json file extracted through the backend as well as the netron function.

**Comment by seanshpark** [comment]
> Implementation of visualization like Netron

Why do we have to implement like netron from scratch?

> comparing the json,

Are the json files you mention are https://github.com/Samsung/ONE-vscode/issues/8#issuecomment-905421976 ?

**Comment by CODe1995** [comment]
> Are the json files you mention are #8 (comment) ?  

here [#66 (comment)](https://github.com/Samsung/ONE-vscode/issues/66#issuecomment-916629684)

![image](https://user-images.githubusercontent.com/47562093/134660541-63c8068e-5284-4092-97c1-9daef2182429.png)
This is a function that compares two jsons, shows the difference in `duration`, and highlights the section where the bottleneck occurs. I'm trying to show this intuitively like a graph of `netron`.

**Comment by seanshpark** [comment]
> jsons, shows the difference in duration, and highlights the section 

OK, thanks for the explanation, it's `like` netron (show nodes in graph) but we need different presentation.

**Comment by CODe1995** [comment]
> we need different presentation.

Do you mean change the UI?

**Comment by seanshpark** [comment]
> Do you mean change the UI?

I don't exactly understand the question but, I just wrote `why we need another view like netron?` as `but we need different presentation.`

**Comment by seanshpark** [comment]
As I understand you need a solution to access circle model graph from webview.
- that is something like get data from nodejs server from browser js side.

Have you looked into https://github.com/Microsoft/vscode-extension-samples/tree/main/webview-sample ?
As I understand this can be a solution.
That is,
we provide flatbuffer reading and stuff in extension code,
and webview JS requests to extension code to get the data.

**Comment by CODe1995** [comment]
We've already written code based on that file.

There is no problem in using the js file itself. However, there is a problem with the use of `require` in js.

![image](https://user-images.githubusercontent.com/47562093/134663642-1340fb14-4f08-4881-953b-aad2065a98f1.png)

Even vscode recommends using `bundler`.
In other words, when combined into one file, 8800 lines of code are created. In this case, I am curious how to make a `Pull Request`. The code contains dozens of `circle flatbuffers` files in addition to our code.

**Comment by seanshpark** [comment]
>  I am curious how to make a Pull Request. The code contains dozens of circle flatbuffers files in addition to our code.

What I could say is that this is not recommended and hope you find a way with WebView API.

**Comment by CODe1995** [comment]
Thanks, I'll find a way.
